x-pack/plugins/alerting/server/rules_client/rules_client.ts

```diff
@@ -1225,6 +1307,231 @@ export class RulesClient {
+          error,
+        })
+      );
+      throw error;
+    }
+    const { filter: authorizationFilter } = authorizationTuple;
+    const qNodeFilter =
+      (authorizationFilter && filter
+        ? nodeBuilder.and([
+            typeof filter === 'string' ? fromKueryExpression(filter) : filter,
+            authorizationFilter as KueryNode,
+          ])
+        : authorizationFilter) ?? filter;
+
+    const { aggregations, total } = await this.unsecuredSavedObjectsClient.find<
+      RawRule,
+      RuleBulkEditAggregation
+    >({
+      filter: qNodeFilter,
+      page: 1,
+      perPage: 0,
+      type: 'alert',
+      aggs: {
+        alertTypeId: {
+          multi_terms: {
+            terms: [
+              { field: 'alert.attributes.alertTypeId' },
+              { field: 'alert.attributes.consumer' },
+            ],
+          },
+        },
+      },
+    });
+
+    if (total > MAX_RULES_NUMBER_FOR_BULK_EDIT) {
+      throw Error(`More than ${MAX_RULES_NUMBER_FOR_BULK_EDIT} rules matched for bulk edit`);
+    }
+    const buckets = aggregations?.alertTypeId.buckets;
+
+    if (buckets === undefined || buckets.length === 0) {
+      throw Error('Rules not found');
+    }
+
+    await pMap(buckets, async ({ key: [ruleType, consumer] }) => {
+      this.ruleTypeRegistry.ensureRuleTypeEnabled(ruleType);
+
+      await this.authorization.ensureAuthorized({
+        ruleTypeId: ruleType,
+        consumer,
+        operation: WriteOperations.Update,
+        entity: AlertingAuthorizationEntity.Rule,
+      });
+    });
+
+    const rulesFinder =
+      await this.encryptedSavedObjectsClient.createPointInTimeFinderAsInternalUser<RawRule>(
+        { namespace: this.namespace, type: 'alert' },
+        {
+          filter,
+          type: 'alert',
+          perPage: 1000,
+        }
+      );
+
+    const rules: Array<SavedObjectsBulkUpdateObject<RawRule>> = [];
+    const errors: BulkEditError[] = [];
+    const apiKeysToInvalidate: string[] = [];
+
+    for await (const response of rulesFinder.find()) {
+      await pMap(
+        response.saved_objects,
+        async (rule) => {
+          try {
+            if (rule.attributes.apiKey) {
+              apiKeysToInvalidate.push(rule.attributes.apiKey);
+            }
+
+            const ruleType = this.ruleTypeRegistry.get(rule.attributes.alertTypeId);
+
+            let attributes = cloneDeep(rule.attributes);
+            let ruleActions = {
+              actions: this.injectReferencesIntoActions(
+                rule.id,
+                rule.attributes.actions,
+                rule.references || []
+              ),
+            };
+            for (const action of actions) {
+              if (action.field === 'actions') {
+                await this.validateActions(ruleType, action.value);
+                ruleActions = this.applyBulkEditAction(action, ruleActions);
+              }
+
+              attributes = this.applyBulkEditAction(action, attributes);
```

Comment:
Forget `else`
```suggestion
     else {
              attributes = this.applyBulkEditAction(action, attributes);
    }
```

Reply:
converted to switch https://github.com/elastic/kibana/pull/126904#discussion_r834609489